### PR TITLE
Remove merged from kafka binding configuration

### LINKED ISSUE #108: Remove merged from kafka binding configuration
Cleanup `merged` in configs, remove from schema and online docs.

### PR DESCRIPTION
**Pull request by ankit-kumar-25**
Fix #108

### CODE REVIEW

File: runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java [hunk 1]
```diff
@@ -204,42 +204,37 @@ public MessageConsumer newStream(
...
-                    authorization,
-                    topicName,
-                    resolvedId,
-                    capabilities,
-                    initialOffsetsById,
-                    defaultOffset,
-                    isolation,
-                    deltaType,
-                    ackMode)::onMergedMessage;
-        }
```
Comment (jfallows) [inline runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java:242]: This change loses the `binding != null` condition too.

**Review by jfallows — CHANGES_REQUESTED** [review]
Please also consider schema validation and the potential downstream impact on https://github.com/aklivity/zilla-examples/ and https://docs.aklivity.io/zilla/get-started configurations.

File: runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java (hunk 1 above)
Comment (ankit-kumar-25) [reply runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java:242]: `binding` is not getting used at any other place, so I wasn’t sure if a null check is required.

**Comment by ankit-kumar-25** [comment]
For schema validation and the potential downstream impact, I was think of introducing another constructor for KafkaOptionsConfig with merged attritbute and make it deprecated. So that we have backward compatibility.
@jfallows Can you please share insights on this?

**Comment by jfallows** [comment]
> For schema validation and the potential downstream impact, I was think of introducing another constructor for KafkaOptionsConfig with merged attritbute and make it deprecated. So that we have backward compatibility. @jfallows Can you please share some insights on this?

@ankit-kumar-25 Yes, that's a good idea to deprecate the `merged` option initially, to minimize backwards incompatibility in the short term and schedule it for removal later.

Our contract with the users of zilla is the `zilla.json` configuration, not the internal code APIs, so for users to see the effect of such deprecation, it would have to be applied to JSON schema validation.

Take a look at https://github.com/aklivity/zilla/tree/develop/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/schema and let me know your thoughts.

Suggest also removing all occurrences of `merged` `kafka` binding option from any `zilla-examples` currently specifying it explicitly. This will involve a separate PR on `zilla-examples` repository.

**Comment by ankit-kumar-25** [comment]
@jfallows Update the Schema with deprecated flag for `merged` attribute. 

Also raised PR: https://github.com/aklivity/zilla-examples/pull/11 to remove all occurrences of merged kafka binding option in zilla-examples.

Can you please review this? Thanks!

File: runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java (hunk 1 above)
Comment (ankit-kumar-25) [reply runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java]: done, reverted the `binding != null` condition.

**Review by jfallows — APPROVED** [review]

**Pull request merged**